Fail
file path is :
projects/deltachat-core/rust/src/chat.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error: cannot find macro `paramsv` in this scope
    --> src/chat.rs:3942:13
     |
3942 |             paramsv![chat_id, contact_id],
     |             ^^^^^^^ help: a macro with a similar name exists: `params`
     |
    ::: /home/k/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rusqlite-0.31.0/src/lib.rs:176:1
     |
176  | macro_rules! params {
     | ------------------- similarly named macro `params` defined here

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3901:27
     |
3901 |         || (contact_id <= DC_CONTACT_ID_SELF && contact_id != DC_CONTACT_ID_SELF)
     |                           ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3901:63
     |
3901 |         || (contact_id <= DC_CONTACT_ID_SELF && contact_id != DC_CONTACT_ID_SELF)
     |                                                               ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `msg_del_member_local` in module `message`
    --> src/chat.rs:3923:22
     |
3923 |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
     |                      ^^^^^^^^^^^^^^^^^^^^ not found in `message`
     |
help: consider importing this function
     |
3    + use crate::stock_str::msg_del_member_local;
     |
help: if you import `msg_del_member_local`, refer to it directly
     |
3923 -             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
3923 +             msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
     |

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
    --> src/chat.rs:3923:67
     |
3923 |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
     |                                                                   ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0599]: no method named `is_null` found for reference `&context::Context` in the current scope
    --> src/chat.rs:3899:16
     |
3899 |     if context.is_null()
     |                ^^^^^^^ method not found in `&Context`

error[E0599]: no method named `is_null` found for struct `Chat` in the current scope
    --> src/chat.rs:3910:13
     |
1480 | pub struct Chat {
     | --------------- method `is_null` not found for this struct
...
3910 |     if chat.is_null() {
     |             ^^^^^^^ method not found in `Chat`

error[E0599]: no method named `is_self_in_group` found for struct `Chat` in the current scope
    --> src/chat.rs:3914:14
     |
1480 | pub struct Chat {
     | --------------- method `is_self_in_group` not found for this struct
...
3914 |     if !chat.is_self_in_group() {
     |              ^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `is_self_in_chat`

error[E0616]: field `addr` of struct `Contact` is private
    --> src/chat.rs:3923:61
     |
3923 |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
     |                                                             ^^^^ private field

error[E0308]: mismatched types
    --> src/chat.rs:3922:26
     |
3922 |           if !msg.set_text(Some(
     |  _________________--------_^
     | |                 |
     | |                 arguments to this method are incorrect
3923 | |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
3924 | |         )) {
     | |_________^ expected `String`, found `Option<_>`
     |
     = note: expected struct `std::string::String`
                  found enum `std::option::Option<_>`
note: method defined here
    --> src/message.rs:1048:12
     |
1048 |     pub fn set_text(&mut self, text: String) {
     |            ^^^^^^^^            ------------

error[E0600]: cannot apply unary operator `!` to type `()`
    --> src/chat.rs:3922:12
     |
3922 |           if !msg.set_text(Some(
     |  ____________^
3923 | |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
3924 | |         )) {
     | |__________^ cannot apply unary operator `!`

error[E0616]: field `addr` of struct `Contact` is private
    --> src/chat.rs:3929:43
     |
3929 |         msg.param.set(Param::Arg, contact.addr);
     |                                           ^^^^ private field

error[E0599]: no method named `is_null` found for struct `chat::Chat` in the current scope
    --> src/chat.rs:3910:13
     |
1480 | pub struct Chat {
     | --------------- method `is_null` not found for this struct
...
3910 |     if chat.is_null() {
     |             ^^^^^^^ method not found in `chat::Chat`

error[E0599]: no method named `is_self_in_group` found for struct `chat::Chat` in the current scope
    --> src/chat.rs:3914:14
     |
1480 | pub struct Chat {
     | --------------- method `is_self_in_group` not found for this struct
...
3914 |     if !chat.is_self_in_group() {
     |              ^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `is_self_in_chat`

error[E0616]: field `addr` of struct `contact::Contact` is private
    --> src/chat.rs:3923:61
     |
3923 |             message::msg_del_member_local(context, &contact.addr, DC_CONTACT_ID_SELF).await,
     |                                                             ^^^^ private field

error[E0616]: field `addr` of struct `contact::Contact` is private
    --> src/chat.rs:3929:43
     |
3929 |         msg.param.set(Param::Arg, contact.addr);
     |                                           ^^^^ private field

Some errors have detailed explanations: E0308, E0425, E0599, E0600, E0616.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `deltachat` (lib) due to 12 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `deltachat` (lib test) due to 12 previous errors
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

